Userland/Utilities/cp.cpp

```diff
@@ -74,11 +74,8 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     bool destination_is_existing_dir = FileSystem::is_directory(destination);
 
     for (auto& source : sources) {
-        // FIXME: May be formatted wrong if destination is a directory with a trailing slash, e.g. if called as such:
-        // cp source dest/
-        // ... `destination_path` may be formatted as `dest//source`, which isn't strictly correct.
         auto destination_path = destination_is_existing_dir
-            ? ByteString::formatted("{}/{}", destination, LexicalPath::basename(source))
+            ? LexicalPath::canonicalized_path(ByteString::formatted("{}/{}", destination, LexicalPath::basename(source)))
```

Comment:
This will resolve symlinks, which at least other `cp` implementations don't do in their `-v output:

```
% mkdir -p dir1/dir2/dir3
% ln -s dir1/dir2/dir3 dir1/symlink
% touch foo.txt
% cp -v foo.txt dir1/symlink/..         
foo.txt -> dir1/symlink/../foo.txt
```

I think this will probably print `foo.txt -> dir1/foo.txt` instead (…but I haven't tried it).

Reply:
It prints `'foo.txt' -> 'dir1/foo.txt'`.